burnman/anisotropy.py

```diff
@@ -0,0 +1,449 @@
+                           [(1, 1)], # C22
+                           [(1, 2)], # C23
+                           [(2, 2)], # C33
+                           [(3, 3)], # C44
+                           [(4, 4)], # C55
+                           [(5, 5)]] # C66
+            
+        elif crystal_system=='monoclinic':
+            assert len(cijs) == 13
+            index_lists = [[(0, 0)], # C11
+                           [(0, 1)], # C12
+                           [(0, 2)], # C13
+                           [(0, 4)], # C15
+                           [(1, 1)], # C22
+                           [(1, 2)], # C23
+                           [(1, 4)], # C25
+                           [(2, 2)], # C33
+                           [(2, 4)], # C35
+                           [(3, 3)], # C44
+                           [(3, 5)], # C46
+                           [(4, 4)], # C55
+                           [(5, 5)]] # C66
+            
+        elif crystal_system=='triclinic':
+            assert len(cijs) == 21
+            index_lists=[[(i, j)] for i in range(6) for j in range(i, 6)]
+            
+        else:
+            raise Exception('Crystal system not recognised. Must be one of: '
+                            'isotropic, cubic, hexagonal, tetragonal I, '
+                            'tetragonal II, rhombohedral I, rhombohedral II, '
+                            'orthorhombic, monoclinic or triclinic.')    
+            
+        
+        C = np.zeros([6, 6])
+        for i, index_list in enumerate(index_lists):
+            for indices in index_list:
+                C[indices] = cijs[i]
+                C[indices[::-1]] = cijs[i]
+        return C
+        
+    def _voigt_index_to_ij(self, m):
+        """
+        Returns the ij (or kl) indices of the 
+        stiffness tensor which correspond to those 
+        of the Voigt notation m (or n).
+        """
+        if m == 3:
+            i=1
+            j=2
+        elif m == 4:
+            i=0
+            j=2
+        elif m == 5:
+            i=0
+            j=1
+        else:
+            i=m
+            j=m
+        return i, j
+
+    def _normalize(self, vector):
+        v = np.array(vector)
+        return v/np.linalg.norm(v) 
+
+    def _voigt_notation_to_stiffness_tensor(self, voigt_notation):
+        """
+        Converts a stiffness tensor in Voigt notation (6x6 matrix)
+        to the full fourth rank tensor (3x3x3x3 matrix).
+        """
+        stiffness_tensor = np.zeros([3, 3, 3, 3])
+        for m in range(6):
+            i, j = self._voigt_index_to_ij(m)
+            for n in range(6):
+                k, l = self._voigt_index_to_ij(n)
+                stiffness_tensor[i][j][k][l] = voigt_notation[m][n]
+                stiffness_tensor[j][i][k][l] = voigt_notation[m][n]
+                stiffness_tensor[i][j][l][k] = voigt_notation[m][n]
+                stiffness_tensor[j][i][l][k] = voigt_notation[m][n]
+        return stiffness_tensor
+    
+    @property
```

Comment:
@tjhei I think there are some communication issues here. I meant that unless I ask for material.bulk_modulus_voigt, it won't be calculated. Is that correct? 

@tjhei and @sannecottaar , the points you make about caching, and more generally deriving from Material are good ones. I've reworked AnisotropicMaterial to derive from Material. Eventually, it would be good to add a set_state, but there isn't any point at the moment, as we don't have an appropriate equation of state to use. (Coding up the full SLB EoS might be a nice project to give to a student, but is way outside the scope of this PR).